ext/src/save.rs

```diff
@@ -399,6 +433,17 @@ impl<A: Algebra> SaveFile<A> {
         Ok(())
     }
 
+    /// Whether we should load the file in memory and check the checksum before returning it. This
+    /// only returns false for quasi-inverses because they are our largest files by far. This is a
+    /// function of `SaveFile` and not just `SaveKind` because we may want to change the behavior
+    /// depending on the stem or some other heuristic.
+    fn should_check_early(&self) -> bool {
+        !matches!(
+            self.kind,
+            SaveKind::AugmentationQi | SaveKind::NassauQi | SaveKind::ResQi
+        )
+    }
```

Comment:
This makes sense as an improvement over the status quo but there may be a better design of the file format that would allow checking only the section of the file that is used? For instance you could place a checksum every 4kb or imitate the zip file format. Another possibility is there might be an existing container format that is appropriate and that could make the checksums transparent to our code.

Reply:
I looked around and didn't see a generic container that would do that for us, but one interesting option would be to always output zstd-compressed data. Zstd has options to do frame-level and block-level checksumming, and we already support reading compressed files.

Also, I just checked with our stem 200 data and, apart from the quasi-inverses, the biggest files only take a few MB, with the vast majority taking less than 1 kB. In fact, the first thing that the code does after opening those files is reading the entire contents, so we're really just prefetching